### PR TITLE
Sign in integration 

### PR DESCRIPTION
**Pull request by rahma-nagy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "@mui/material": "^5.15.2",
     "@mui/styled-engine": "^5.15.2",
     "axios": "^1.6.3",
+    "formik": "^2.4.5",
```
Comment (Ahmed-Abou-Emran) [inline package.json]: You are using react-hook-form, so you need to uninstall formik to reduce the bundle size.

File: src/App.tsx [hunk 2]
```diff
@@ -18,13 +18,19 @@ import User from './Components/User/User'
...
 
 function App() {
+ 
+  let {userData,saveUserData,role}:any=useContext(AuthContext)
 const routes =createBrowserRouter([
   {
     path:"/",element:<AuthLayout/>,errorElement:<NotFound/>,
     children:[
       {index:true,element:<LandingPage/>},
+      {path:"nav",element:<Navbar/>},
```
Comment (Ahmed-Abou-Emran) [inline src/App.tsx]: Navbar is part of the AppLayout but it is not a separate page with a distinct path.

File: src/Context/AuthContext.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+export default function AuthContextProvider(props:any){
+
+  const [userData, setUserData] = useState(null);
+  const[role,setRole]=useState(null)
+  
+
+  let reqHeaders={
+    Authorization:`Bearer ${localStorage.getItem("userToken")}`
```
Comment (Ahmed-Abou-Emran) [inline src/Context/AuthContext.tsx:16]: You 'll need to remove `Bearer` here as the token coming from the backend is `Bearer TokenValue`, so if you used the current token the request header will be `Bearer Bearer TokenValue`

image

File: src/Context/AuthContext.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  }
+  const baseUrl = "http://upskilling-egypt.com:3000/api/v0";
+
+  const saveUserData = () => {
+    const encodedToken:any = localStorage.getItem("userToken");
+    const decodedToken:any = jwtDecode(encodedToken)
```
Comment (Ahmed-Abou-Emran) [inline src/Context/AuthContext.tsx:22]: We don't need to use jwtDecode, as the data coming from the backend when we login is what we get from the decoded token.

**Review by Ahmed-Abou-Emran — CHANGES_REQUESTED** [review]
### Required Changes:

- [x] Removing unused packages.
- [x] Removing unused fonts.
- [x] Removing logs.
- [x] Removing `NavBar` path from the routing.
- [x] Removing `Bearer` as it is already included in the token coming from the backend.
- [x] After completing the previous points, there is a conflict in the branch with main branch, we need to fix that to be able to merge the code to the main.

File: src/App.tsx (hunk 2 above)
Comment (rahma-nagy) [reply src/App.tsx]: hwa 3shan fe 2 navbar f hawelt a5tre3 :'D

File: package.json (hunk 1 above)
Comment (rahma-nagy) [reply package.json]: SORRY :'D

**Pull request merged**